Packs/McAfee-TIE/Integrations/McAfee-TIE_V2/McAfee-TIE_V2.py

```diff
@@ -0,0 +1,603 @@
+import abc
+import contextlib
+import tempfile
+from CommonServerPython import *
+from dxlclient.client_config import DxlClientConfig
+from dxlclient.client import DxlClient
+from dxlclient.broker import Broker
+from dxltieclient import TieClient
+from typing import NamedTuple
+from dxltieclient.constants import FileReputationProp, FileGtiAttrib, FileEnterpriseAttrib, AtdAttrib, TrustLevel,\
+    HashType, EnterpriseAttrib, FileProvider, FirstRefProp, AtdTrustLevel
+
+VENDOR_NAME = 'McAfee Threat Intelligence Exchange'
+OUTPUT_PREFIX = 'McAfee.TIE'
+LOWEST_TRUST_LEVEL_KEY = 'lowest_trust_level_key'
+LOWEST_SCORE_KEY = 'lowest_score_key'
+MAX_QUERY_LIMIT = 500
+
+DXLConfigFiles = NamedTuple('DXLConfigFiles', [('broker_ca_bundle', str),
+                                               ('cert_file', str),
+                                               ('private_key', str),
+                                               ('broker_urls', list[str]),
+                                               ])
+
+ProviderInfo = NamedTuple('ProviderInfo', [('name', str),
+                                           ('abbreviation', str)
+                                           ])
+
+
+PROVIDER_INFO = {
+    FileProvider.GTI: ProviderInfo(name='Global Threat Intelligence (GTI)', abbreviation='GTI'),
+    FileProvider.ENTERPRISE: ProviderInfo(name='Enterprise reputation', abbreviation='Enterprise'),
+    FileProvider.ATD: ProviderInfo(name='Advanced Threat Defense (ATD)', abbreviation='ATD'),
+    FileProvider.MWG: ProviderInfo(name='Web Gateway (MWG)', abbreviation='MWG'),
+}
+
+
+TRUST_LEVELS = {
+    TrustLevel.KNOWN_TRUSTED_INSTALLER: 'KNOWN_TRUSTED_INSTALLER',
+    TrustLevel.KNOWN_TRUSTED: 'KNOWN_TRUSTED',
+    TrustLevel.MOST_LIKELY_TRUSTED: 'MOST_LIKELY_TRUSTED',
+    TrustLevel.MIGHT_BE_TRUSTED: 'MIGHT_BE_TRUSTED',
+    TrustLevel.UNKNOWN: 'UNKNOWN',
+    TrustLevel.MIGHT_BE_MALICIOUS: 'MIGHT_BE_MALICIOUS',
+    TrustLevel.MOST_LIKELY_MALICIOUS: 'MOST_LIKELY_MALICIOUS',
+    TrustLevel.KNOWN_MALICIOUS: 'KNOWN_MALICIOUS',
+    TrustLevel.NOT_SET: 'NOT_SET',
+}
+
+HASH_TYPE_KEYS = {
+    'md5': HashType.MD5,
+    'sha1': HashType.SHA1,
+    'sha256': HashType.SHA256
+}
+
+
+class GeneralFileReputationParser(abc.ABC):
```

Comment:
Please provide a docstring for the class, and its methods (espacially the abstract ones)